primitive/zk_rollup/src/poseidon.rs

```diff
@@ -3,18 +3,20 @@
 
 use anyhow::Result;
 use sp_std::marker::PhantomData;
-use zkstd::common::FftField;
+use zkstd::common::{Decode, Encode, FftField};
 
-#[derive(Debug, Clone, PartialEq, Copy)]
-pub struct Poseidon<F: FftField, const L: usize>(PhantomData<F>);
+#[derive(Debug, Clone, PartialEq, Eq, Copy, Encode, Decode)]
+pub struct Poseidon<Fld: FftField, const L: usize>(PhantomData<Fld>);
```

Comment:
Can we change Fld -> F?

Reply:
There was an error about same generic name, when I tried to derive Encode. That's why I changed it

Reply:
Okay
Let's stay the same